File: src/org/usfirst/frc5293/input/util/NullJoystick.java
package org.usfirst.frc5293.input.util;

import edu.wpi.first.wpilibj.GenericHID;
import edu.wpi.first.wpilibj.Joystick;

public final class NullJoystick extends Joystick {

    private NullJoystick() {
        super(0, 0, 0);
    }

    public static final NullJoystick INSTANCE = new NullJoystick();

    @Override
    public double getX(Hand hand) {
        return 0;
    }

    @Override
    public double getY(Hand hand) {
        return 0;
    }

    @Override
    public double getZ(Hand hand) {
        return 0;
    }

    @Override
    public double getTwist() {
        return 0;
    }

    @Override
    public double getThrottle() {
        return 0;
    }

    @Override
    public double getRawAxis(int axis) {
        return 0;
    }

    @Override
    public double getAxis(AxisType axis) {
        return 0;
    }

    @Override
    public int getAxisCount() {
        return 0;
    }

    @Override
    public boolean getTrigger(Hand hand) {
        return false;
    }

    @Override
    public boolean getTop(Hand hand) {
        return false;
    }

    @Override
    public boolean getBumper(Hand hand) {
        return false;
    }

    @Override
    public boolean getRawButton(int button) {
        return false;
    }

    @Override
    public int getButtonCount() {
        return 0;
    }

    @Override
    public int getPOV(int pov) {
        return 0;
    }

    @Override
    public int getPOVCount() {
        return 0;
    }

    @Override
    public boolean getButton(ButtonType button) {
        return false;
    }

    @Override
    public double getMagnitude() {
        return 0;
    }

    @Override
    public double getDirectionRadians() {
        return 0;
    }

    @Override
    public double getDirectionDegrees() {
        return 0;
    }

    @Override
    public int getAxisChannel(AxisType axis) {
        return 0;
    }

    @Override
    public void setAxisChannel(AxisType axis, int channel) {
    }

    @Override
    public void setRumble(RumbleType type, float value) {
    }

    @Override
    public void setOutput(int outputNumber, boolean value) {
    }

    @Override
    public void setOutputs(int value) {
    }

    @Override
    public int getPOV() {
        return 0;
    }
}

File: src/org/usfirst/frc5293/input/util/NullJoystick.kt
package org.usfirst.frc5293.input.util

import edu.wpi.first.wpilibj.GenericHID
import edu.wpi.first.wpilibj.Joystick

object NullJoystick : Joystick(0, 0, 0) {

    override fun getX(hand: GenericHID.Hand?): Double {
        return 0.0
    }

    override fun getY(hand: GenericHID.Hand?): Double {
        return 0.0
    }

    override fun getZ(hand: GenericHID.Hand?): Double {
        return 0.0
    }

    override fun getTwist(): Double {
        return 0.0
    }

    override fun getThrottle(): Double {
        return 0.0
    }

    override fun getRawAxis(axis: Int): Double {
        return 0.0
    }

    override fun getAxis(axis: Joystick.AxisType): Double {
        return 0.0
    }

    override fun getAxisCount(): Int {
        return 0
    }

    override fun getTrigger(hand: GenericHID.Hand?): Boolean {
        return false
    }

    override fun getTop(hand: GenericHID.Hand?): Boolean {
        return false
    }

    override fun getBumper(hand: GenericHID.Hand?): Boolean {
        return false
    }

    override fun getRawButton(button: Int): Boolean {
        return false
    }

    override fun getButtonCount(): Int {
        return 0
    }

    override fun getPOV(pov: Int): Int {
        return 0
    }

    override fun getPOVCount(): Int {
        return 0
    }

    override fun getButton(button: Joystick.ButtonType): Boolean {
        return false
    }

    override fun getMagnitude(): Double {
        return 0.0
    }

    override fun getDirectionRadians(): Double {
        return 0.0
    }

    override fun getDirectionDegrees(): Double {
        return 0.0
    }

    override fun getAxisChannel(axis: Joystick.AxisType): Int {
        return 0
    }

    override fun setAxisChannel(axis: Joystick.AxisType, channel: Int) {
    }

    override fun setRumble(type: Joystick.RumbleType, value: Float) {
    }

    override fun setOutput(outputNumber: Int, value: Boolean) {
    }

    override fun setOutputs(value: Int) {
    }

    override fun getPOV(): Int {
        return 0
    }
}
